test/scripts/run_mocha.js

```diff
@@ -0,0 +1,16 @@
+const debugInBand = process.execArgv.some(arg => {
+  switch (arg) {
+    case '--debug':
+    case '--debug-brk':
+    case '-d':
+    case '--inspect':
+      return true;
+  }
+});
+
+if (debugInBand) {
+  process.argv.push('--no-timeouts');
+  require('mocha/bin/_mocha');
```

Comment:
`bin/_mocha` runs the mocha tests in the same process, while the standard mocha CLI runs the tests in a child process. because of the child process we can't `--inspect`, so this switches over to `_mocha` in that scenario. Using `_mocha` has some disadvantages when it comes to error reporting, but I'm thinking that they will be greatly overpowered by having a real debugger.